app/views/unattended/report_templates/jobs_-_invocation_report_template.erb

```diff
@@ -30,15 +30,11 @@ require:
 <%- load_hosts(search: search).each do |batch| -%>
 <%-   batch.each do |host|  -%>
 <%-     task = invocation.sub_task_for_host(host) -%>
+<%-     outputs = { 'stdout' => [], 'stderr' => [], 'debug' => [] }  -%>
```

Comment:
Not sure if there's anything we can do about it, but I think this might significantly increase the memory usage when generating a report for invocations that had very long outputs.

Reply:
It was a long array of short strings, now it's a smaller array of longer strings. The amount of characters though remains the same. The outputs hash is cleared for every job and is not linked from anywhere, so GC should deallocate that. Or did you mean some other part?

Reply:
i don't recall the internals of `report_row`, don't we stream/write it one row at a time? so instead of loading one output line every iteration and writing it, we load all the outputs to memory before writing them all in one go.

Reply:
We don't stream that, we're storing this in DB and allow to download that rendered report later. So we load everything into memory, then store it to DB. When it's downloaded, we load the whole thing from PG to Rails process and send the file. Streaming was never finished https://github.com/theforeman/foreman/pull/8136